qhana_plugin_registry/api/models/templates.py

```diff
@@ -89,7 +89,7 @@ def validate_filter(filter_dict: dict, path: str = ""):
                         f"Invalid plugin filter: Invalid version '{v}'. Version must be a PEP 440 specifier (string). (Path: {current_path})"
                     )
                 try:
-                    SpecifierSet(v)
+                    Specifier(v)
```

Comment:
Why not keep `SpecifierSet` here for added flexibility in specifying allowed versions? (Also: try to mention all relevant changes in the PR text, I would consider this one relevant for this PR)

Reply:
Reasons for this change:
    - The plugin filter key 'version' suggests that only one version can be specified.
    - The `SpecifierSet` implementation could break the current test setup. Hypothesis might generate a test case with a version that's a valid `Specifier` but an invalid `SpecifierSet` (e.g. `'=====,0'`).

**I'll revert this change and adjust the unit test to work with `SpecifierSet`s.**

Reply:
- in this context the "version" is used as a filter (where one usually wants to filter by version ranges and such)
- That are edge cases you will only find with such thorough tests. So 👍 for the test cases so far. If invalid specifier sets are rare, then you can filter those out after generating them with [assume](https://hypothesis.readthedocs.io/en/latest/details.html#making-assumptions)